eth2near/eth2near-block-relay-rs/src/eth2near_relay.rs

```diff
@@ -19,22 +22,28 @@ pub struct Eth2NearRelay {
     current_gap_between_finalized_and_signature_slot: u64,
     network: String,
     light_client_updates_submission_frequency_in_epochs: i64,
+    max_blocks_for_finalization: u64,
+    enable_binsearch: bool,
 }
 
 impl Eth2NearRelay {
-    pub fn init(config: &Config, contract_wrapper: Box<dyn ContractWrapper>) -> Self {
+    pub fn init(config: &Config, contract_wrapper: Box<dyn ContractWrapper>, enable_binsearch: bool) -> Self {
         info!(target: "relay", "=== Relay initialization === ");
 
         let eth2near_relay = Eth2NearRelay {
             beacon_rpc_client: BeaconRPCClient::new(&config.beacon_endpoint),
             eth1_rpc_client: Eth1RPCClient::new(&config.eth1_endpoint),
             eth_client_contract: EthClientContract::new(contract_wrapper),
             max_submitted_headers: config.total_submit_headers as u64,
-            current_gap_between_finalized_and_signature_slot: 96,
+            current_gap_between_finalized_and_signature_slot: Self::get_gap_between_finalized_and_signature_slot(
+                config.light_client_updates_submission_frequency_in_epochs as u64),
             network: config.network.to_string(),
             light_client_updates_submission_frequency_in_epochs: config
                 .light_client_updates_submission_frequency_in_epochs,
+            max_blocks_for_finalization: config.max_blocks_for_finalization,
+            enable_binsearch
         };
+
         eth2near_relay
             .eth_client_contract
             .register_submitter()
```

Comment:
We should avoid call `register_submitter` on every start, to do that we should add an API to the contract

Reply:
Added flag for now https://github.com/aurora-is-near/rainbow-bridge/pull/775/commits/3a5d0bd63c314c01b6b0d473bc4361d9d599c2d9

Reply:
Created an issue https://github.com/aurora-is-near/rainbow-bridge/issues/777#issue-1327852192